src/visualization/utilities.jl

```diff
@@ -859,6 +860,113 @@ function unstructured_3d_to_1d_curve(original_nodes, unstructured_data, nvisnode
   return arc_length, data_on_curve, nothing
 end
 
+# Convert 3d unstructured data from a general mesh to 1d data at given curve.
+function unstructured_3d_to_1d_curve(nodes, data, curve, slice, point, nvisnodes)
+    if curve === nothing
```

Comment:
Please add a few comments as above, e.g., `# If no curve is defined, create a axis curve.`